src/index.js

```diff
@@ -61,8 +62,9 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
     // `noRouterUpdate` flag can be set to avoid updating altogether,
     // which is useful for things like loading snapshots or very special
     // edge cases.
-    if(routing.path !== locationToString(window.location) &&
+    if(lastRoute !== routing.path && routing.path !== locationToString(window.location) &&
```

Comment:
Of course, I'll add a comment. Please let me know if my assumption about the `true` in `updatePath` is correct